drivers/serial/Kconfig

```diff
@@ -28,12 +28,20 @@ config SERIAL_SUPPORT_INTERRUPT
 	  This is an option to be enabled by individual serial driver
 	  to signal that the driver and hardware supports interrupts.
 
+config UART_HIGH_LEVEL_API
+	bool "Enable new high level UART API [EXPERIMENTAL]"
+	help
+	  This option enables new UART API, which will become standard in the
+	  future.
+
+if UART_HIGH_LEVEL_API=n
```

Comment:
I don't think it is possible to use both with same UART instance. There could be a way to use them separately for different instances, but it would have to be enforced on driver side not on API